### PR TITLE
Use Scout as TradeLeader

### PR DESCRIPTION
**Pull request by Shawn8901**
I have implemented the request from #218.

Additionally i have changed two things:
* i did not understand, why Helpers limits the trade for Figures, since they are not reduced if caravan leaves, i removed this limitation (maybe i am missing something?)
* removed the military recruitment "simulation" from test case, since there is no dependency left to Helpers

edit: coverage from coverall is missing because of 0664b9b i will rebase, if #773 will be merged before this. i am not sure what happens if i am cherry picking this commit to this branch, will git be able to handle this?

edit2: rebased.

### CODE REVIEW

File: src/buildings/nobBaseWarehouse.cpp [hunk 1]
```diff
@@ -1408,24 +1408,28 @@ bool nobBaseWarehouse::IsDependentFigure(noFigure* fig) const
 /// Available goods of a specific type that can be used for trading
 unsigned nobBaseWarehouse::GetAvailableWaresForTrading(const GoodType gt) const
 {
-    // We need a helper as leader
-    if(!inventory[JOB_HELPER])
+    // We need a scout as leader or a bow to recruit
+    bool noScoutAvailable = !inventory[JOB_SCOUT];
+    if (noScoutAvailable && !inventory[GD_BOW])
```
Comment (Flamefire) [inline src/buildings/nobBaseWarehouse.cpp]: Actually a bow and a helper and you should use `CanRecruit` instead

Comment (Shawn8901) [reply src/buildings/nobBaseWarehouse.cpp]: actually you are right! i will add this.

Other question: 
Do you have access to the lobby server? For me it seems that the lobby is down.

Comment (Flow86) [reply src/buildings/nobBaseWarehouse.cpp]: > Do you have access to the lobby server? For me it seems that the lobby is down.

thats fixed, gitlab upgrade changed some network config stuff

**Comment by Flamefire** [comment]
General question: Do we actually want a scout? What about just using scout graphics? Because he does not need a bow as he just leads the caravan. What would he use the bow for?
ccing @Spikeone

**Comment by Shawn8901** [comment]
in reference to #226 i thought it would be okay to use the scout/bow, because the caravan leader has a higher economic value.

**Comment by Flamefire** [comment]
True. I was just thinking about the implications of sending a (fully equipped) scout.

IMO the PR is ok, but I'd like 1 more opinion on whether to actually use the scout. If yes we can close #226 too

**Comment by Shawn8901** [comment]
i am okay with that, if the intention was only to use the model, please close this pr.
If it should be merged, i have modified the test case for the trade messages to scouts as leader, just in case it will be merged.

**Comment by Spikeone** [comment]
@Flamefire @Shawn8901 As far as I remember the idea was to actually use a scout so caravans cost more. Currently you are able to send soldier after soldier to an ally, boosting him quite a lot. This idea was born when we played 3vs3 and 2 players just sent all soldiers to one player so he could defeat his enemy fast. 

Overall I'm not really sure if this is the best idea as it makes trade even more complicated as it already is for new players. I'd vote for just using the graphic for now and wait till the trade system is reworked - if we still intend to do so. If we will not rework the system, one should have an addon (enable / disable use "real scouts" for trade)

**Comment by Shawn8901** [comment]
Then i would suggest to close/decline the PR, as it was not supposed to be a scout.

**Comment by Flamefire** [comment]
If I understand this correctly the only change required is that the "scout" is taken from the normal people and not from the scouts, right? Could you add this as a separate commit (so we can revert it if we decide FOR using scouts)?

**Comment by Shawn8901** [comment]
i can dig into it and try to find that code passage as soon, as i have reinstalled my dev environment for s25 (had switched from HDD to SSD)

**Comment by Shawn8901** [comment]
I have updated everything to the newest master.
I have commited a single revert commit for the unit change to scout and done a commit to just change the skin.
I hope that its now like expected.

**Comment by Shawn8901** [comment]
checked, also LGTM

**Pull request merged**